### ISSUE TITLE
Need GLSL Shader validation for MaterialX test suite

### ISSUE THREAD

**Issue by ashwinbhat** [AcademySoftwareFoundation/MaterialX#960]
Run the MaterialXTest - GenShader: GLSL Shader Generation test suite. The tests pass because we are testing for successful ShaderGen. However, the shader generated by this test does not compile. We need simple shader validation for MaterialXTest - GenShader: GLSL Shader Generation test suite

e.g. [shaderref0_pixel_genshadertest.glsl.txt](https://github.com/AcademySoftwareFoundation/MaterialX/files/8743545/shaderref0_pixel_genshadertest.glsl.txt)

`glslangValidator.exe -S frag shaderref0_pixel_genshadertest.glsl.txt`
```
ERROR: 0:799: 'L' : redefinition
ERROR: 0:799: '' : compilation terminated
ERROR: 2 compilation errors.  No code generated.

```

**Comment by kwokcb** [comment]
@ashwinbhat , can you add in info as to how you generated this (which file, options). I don't see anything amiss in running render tests against GLSL except for some new regression in image baking ( will log a new issue for this).

**Comment by kwokcb** [comment]
The dependency definitions for lights is using the initial rig set up way way back. 
As we're using "rig 2" for testing I suggest this code be updated; (https://github.com/AcademySoftwareFoundation/MaterialX/blob/4f44b5a0e465ba6ec9abe8f246b504517fa32efa/source/MaterialXTest/MaterialXGenGlsl/GenGlsl.h#L47).

There is something wrong with the compound generation, I assume since it's getting confused since there is an edf node in the graph? It's skipping generation when it should not in addition to duplication. This can be queued to look at but seems low priority. Even non-compound light nodes I'm not sure about the future of ? They we're "examples" as far as I remember.

For generation testing can hookup the glsl fragment tester like we already do with oslc and mdlc and this can run with CI as it's already being installed for a single test there. Ideally the render tests should be runnable in CI at some point ? (:crossed_fingers:)

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @ashwinbhat, and I believe the original issue with code generation for light nodes has been resolved in #984.  If there are related problems still to be resolved, let's plan to create new issues for these on GitHub.
